File: main/com/example/storage/HistoryStorage.java
package com.example.storage;

import com.example.model.OperationModel;
import com.example.model.SimpleOperation;

import java.util.List;

public interface HistoryStorage {
    
    List<String> getHistory(SimpleOperation operationType);
    
    void addToHistory(SimpleOperation operationType, OperationModel operationModel, String result);
    
}

File: main/com/example/storage/HistoryStorage.kt
package com.example.storage

import com.example.model.SimpleOperation
import com.example.model.OperationModel

interface HistoryStorage {
    fun getHistory(operationType: SimpleOperation): List<String>
    fun addToHistory(operationType: SimpleOperation, operationModel: OperationModel, result: String)
}